inline-c/src/Language/C/Inline/Internal.hs

```diff
@@ -298,12 +299,22 @@ inlineCode Code{..} = do
   void $ emitVerbatim $ out $ directive ++ codeDefs
   -- Create and add the FFI declaration.
   ffiImportName <- uniqueFfiImportName
-  dec <- if codeFunPtr
-    then
-      TH.forImpD TH.CCall codeCallSafety ("&" ++ codeFunName) ffiImportName [t| FunPtr $(codeType) |]
-    else TH.forImpD TH.CCall codeCallSafety codeFunName ffiImportName codeType
-  TH.addTopDecls [dec]
-  TH.varE ffiImportName
+  -- Note [ghcide-support]
+  -- haskell-language-server / ghcide cannot handle code that use
+  -- `addForeignFile`/`addForeignSource` as we do here; it will result
```

Comment:
I understand that it doesn't work, but I think what should be fixed is the tool, not the code generation part.
I don't want to add a flag for the tools here.

Reply:
> I think what should be fixed is the tool, not the code generation part.

@junjihashimoto What fix do you suggest? There's none known to me.

If you find it better, we can use an environemnt variable that has in the name that linking does not work in this environment instead of `__GHCIDE__` specifically, so it covers all tools where linking as expected by `inline-c` does not work.

> I don't want to add a flag for the tools here.

Usage of `inline-c` breaks HLS (which it does), everybody will carry this patch anyway.

Reply:
@nh2 One idea is  creating a ghc plugin and adding a mechanism to set functions that don't work.  It's a code generation problem, so I think we can solve a series of problems by the plugin.